numpy/_core/include/numpy/ndarraytypes.h

```diff
@@ -573,6 +573,10 @@ typedef struct {
                                 NPY_ITEM_IS_POINTER | NPY_ITEM_REFCOUNT | \
                                 NPY_NEEDS_INIT | NPY_NEEDS_PYAPI)
 
+#if !(defined(NPY_INTERNAL_BUILD) && NPY_INTERNAL_BUILD)
+/*
+ * Public version of the Descriptor struct
```

Comment:
I'm confused - why does the *public* version still have those things like `names`. Isn't the point that one should use the accessors? Or am I misreading the above `if`?

Reply:
No, you are not:  The point is that I am splitting this into multiple PRs and this one doesn't *actually* change anything.  The only reason I added the release note here, was because Nathan asked for it, and I agreed it might be easier to reason about it here than in a PR that breaks more things.